### PR TITLE
(fix) O3-2756: On clicking the `View` button, the adjacent dropdown menu is triggered, and the dropdown items are displayed. But it should not get displayed. It should be displayed on clicking `Select Service Type` 

### PR DESCRIPTION
**Pull request by subhamkumarr**
## Requirements

- [X] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [X] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [X] My work includes tests or is validated by existing tests.

## Summary
When the user clicks the 'View' button, the dropdown menu located beside it is being triggered unexpectedly. As a result, the dropdown items are shown when they shouldn't be, leading to an unintended user experience.

## Video link:
Before: https://www.loom.com/share/ff73bdb2b6944077bfca4bca29da4663?sid=abac56e1-a135-441c-9d88-f17cb8de2f21

After: https://www.loom.com/share/f7dc119fc1b745b8be697b60c2f8ce65?sid=34b657c2-39f4-4af0-9feb-573227684ab0

## Related Issue
https://openmrs.atlassian.net/browse/O3-2756

### CODE REVIEW

**Review by Pradipram — APPROVED** [review]
nicely done.

File: packages/esm-appointments-app/src/appointments-header/appointments-header.component.tsx [hunk 1]
```diff
@@ -51,24 +51,24 @@ const AppointmentsHeader: React.FC<AppointmentHeaderProps> = ({ title, onChange
             />
           </DatePicker>
         </div>
-        {typeof onChange === 'function' && (
-          <div className={styles.dropdownContainer}>
+        <div className={styles.dropdownContainer}>
```
Comment (sherrif10) [inline packages/esm-appointments-app/src/appointments-header/appointments-header.component.tsx:54]: Well done @subhamkumarr  ,   @denniskigen . Do we  still need the View Button basing on this fix. Looks like it does nothing after this fix.

Comment (subhamkumarr) [reply packages/esm-appointments-app/src/appointments-header/appointments-header.component.tsx:54]: @sherrif10 sir i didn't get you. My PR is not good to merge?

**Comment by sherrif10** [comment]
What i meant, since you are able to click on select service type and its responsive, what is the importance of view button ?

**Comment by subhamkumarr** [comment]
@vasharma05 could you please review my PR

**Comment by subhamkumarr** [comment]
> What i meant, since you are able to click on select service type and its responsive, what is the importance of view button ?

View is not a button it means `view: Select the service `. Just link in `Service Queues` when clicking on `All` dropdown menu is showing

**Comment by vasharma05** [comment]
Please check why the build is failing.
Thanks!

**Comment by subhamkumarr** [comment]
@vasharma05 i never encountered this build error. What should i do? Should i raise another PR for the same issue?

**Comment by vasharma05** [comment]
You can push the changes in the same PR for fixing the build

**Comment by subhamkumarr** [comment]
@vasharma05  by mistakenly I deleted the branch:
I have raised new PR https://github.com/openmrs/openmrs-esm-patient-management/pull/950

**Pull request closed without merging**